includes/wc-core-functions.php

```diff
@@ -731,40 +732,11 @@ function wc_mail( $to, $subject, $message, $headers = "Content-Type: text/html\r
  * @param  string $prop Name of prop (or key::subkey for arrays of props) if you want a specific value. Leave blank to get all props as an array.
  * @param  mixed  $default Optional value to return if the theme does not declare support for a prop.
  * @return mixed  Value of prop(s).
+ *
+ * @deprecated 4.3.0 Use WooCommerce::theme_support()->get_woocommerce_theme_support instead.
```

Comment:
Maybe deprecating isn't the best option, but I think we should somehow encouraging using the new class instead of the function for new code, as code will be easier to test that way.